tests/Mono.Android-Tests/Java.Interop/JavaListTest.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using Android.Runtime;
+using Java.Interop;
+using NUnit.Framework;
+
+namespace Java.InteropTests
+{
+	[Category("JavaList")]
+	[TestFixture(typeof(JavaList))]
+	[TestFixture(typeof(JavaList<string>))]
+	public class JavaListTest<T> where T : JavaList, new()
+	{
+		JavaList list;
+
+		[SetUp]
+		public void Setup () => list = new T ();
+
+		[Test]
+		public void Count ()
+		{
+			list.Add ("foo");
+			Assert.AreEqual (list.Count, 1);
+		}
+
+		[Test]
+		public void Indexer ()
+		{
+			list.Add ("foo");
+			list [0] = "bar";
+			Assert.AreEqual ("bar", list [0]);
+		}
```

Comment:
@jonpryor this is getting close, do you see why this test would fail with:
```
Expected: "bar"
But was:  null
  at Java.InteropTests.JavaListTest`1[T].Indexer () [0x0003a] in <cf4a3c9b3266417699faf3e9acb63b56>:0 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <483bb971b34c48e5aba7c6d09d1efbc7>:0 
```